### PR TITLE
weather-app Sara Stjernberg

### PR DESCRIPTION
**Pull request by stjernberg**
Link to my deployed project:
https://pensive-darwin-b3e353.netlify.app/

My reflections on how this week's project turned out:
 I didn't have time to finish all I wanted to do, du to lack of time this week

Things I'd like to have clarified or explained in more detail:
How to display multiple icons in the fiveday forecast.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    if (e.keyCode === 13) {
+        const inputValue = input.value;
+        cityName = inputValue;
+        const API_URL_FORECAST = `https://api.openweathermap.org/data/2.5/forecast?q=${cityName}&units=metric&APPID=${API_KEY}`;
+        const API_URL_TODAY = `https://api.openweathermap.org/data/2.5/weather?q=${cityName}&units=metric&APPID=${API_KEY}`;
+        getWeatherToday(API_URL_TODAY);
+        getWeatherForecast(API_URL_FORECAST);
+
+    }
+});
```
Comment (mats-johansen) [inline script.js:16]: Great job coding this function!

File: script.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    document.getElementById('degrees').innerHTML = `${currentWeather.main.temp.toFixed(1)} °C`;
+    const description = currentWeather.weather[0].description;
+    document.getElementById('weather-type').innerHTML = description;
+    const weatherPicID = currentWeather.weather[0].icon;
+    weatherPic.src = `./assets/${weatherPicID}.png`;
+
+    //update time of sunrise
+    const sunriseTime = new Date(currentWeather.sys.sunrise * 1000);
+    const sunriseTimeString = sunriseTime.toLocaleTimeString('sv-SE', { timestyle: 'long', hour: "2-digit", minute: "2-digit" });
+    document.getElementById('sunrise').innerHTML += sunriseTimeString;
```
Comment (mats-johansen) [inline script.js:44]: Nice short code for the sunrise- and sunset-time

File: script.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    const dayName = days[d.getDay()];
+    return dayName;
+};
+
+const date = document.getElementById('day')
+date.innerHTML = weekDays();
+
+//get today's date
+const today = () => {
+    const months = ["January", "Febrary", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
```
Comment (mats-johansen) [inline script.js:67]: Nice touch adding the date to the todays forcast!

File: script.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    }
+    //Get today's weather andupdate temperature, weathertype and icon
+const upDateWeather = (currentWeather) => {
+    let city = document.getElementById('name');
+    city.innerHTML = `${currentWeather.name}, ${currentWeather.sys.country}`;
+    document.getElementById('degrees').innerHTML = `${currentWeather.main.temp.toFixed(1)} °C`;
+    const description = currentWeather.weather[0].description;
+    document.getElementById('weather-type').innerHTML = description;
+    const weatherPicID = currentWeather.weather[0].icon;
+    weatherPic.src = `./assets/${weatherPicID}.png`;
```
Comment (mats-johansen) [inline script.js:39]: adding an alt can be good if the picture doesnt work

File: script.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+//get five days forecast
+const getWeatherForecast = (url) => {
+    fetch(url)
+        .then((response) => {
+            return response.json();
+        })
+        .then((forecastArray) => {
+            const filteredArray = forecastArray.list.filter(item => item.dt_txt.includes('12:00'));
+            console.log(filteredArray);
```
Comment (mats-johansen) [inline script.js:83]: remove console.logs when the code is finished

File: script.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+            const forecast = filteredArray.map((forecast) => {
+                const day = (new Date(forecast.dt * 1000)).toLocaleDateString("en-US", { weekday: "long" })
+                const temperature = (forecast.main.temp).toFixed(1);
+                const weatherType = forecast.weather[0].description
+
+                document.getElementById('days').innerHTML += `<p>${day} &nbsp</p>`;
+                document.getElementById('temp').innerHTML += `<p>${temperature} °C  &nbsp &nbsp</p>`;
+                document.getElementById('description').innerHTML += `<p>${weatherType}</p>`;
```
Comment (mats-johansen) [inline script.js:92]: Once you enter a citys name and get the weather for example, Malmö, and then search on Gothenburg you get Gothensburgs weather added onto Malmös weather. So the list on the weekly forcast, sunrise- & sunset-times just add on. I did not to this myself but I am sure you could ask one of the codecoaches how exactly to do this. 

One thing you can test is to just write document.getElementById('temp').innerHTML (WITHOUT THE + SIGN BEFORE THIS -->)= `<p>${temperature} °C  &nbsp &nbsp</p>`;

File: style.css [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+                justify-content: space-around;
+            }
+            .sun {
+                display: flex;
+                justify-content: space-around;
+            }
+            .sun p {
+                margin-right: 50px;
+            }
+        }
```
Comment (mats-johansen) [inline style.css:85]: The styling looks great overall!

File: script.js [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+            return response.json();
+        })
+        .then((forecastArray) => {
+            const filteredArray = forecastArray.list.filter(item => item.dt_txt.includes('12:00'));
+            console.log(filteredArray);
+
+            const forecast = filteredArray.map((forecast) => {
+                const day = (new Date(forecast.dt * 1000)).toLocaleDateString("en-US", { weekday: "long" })
+                const temperature = (forecast.main.temp).toFixed(1);
+                const weatherType = forecast.weather[0].description
```
Comment (mats-johansen) [inline script.js:88]: ;

**Review by mats-johansen — APPROVED** [review]
What I can see you did you did the blue requirements with no problem at all and you also came very far on the red section. I could see a small error when opening your page, there is a small screen on the right hand side but it doesn't do anything do mess with the functions of the weatherapp so it is just as i said, a small error. 

The searchfunction was awesome, great job on that! I commented in your code about the problem that when using the searchfunction, it just keeps addings data ontop of the already fetched data on the dom and it get's kind of messy after a few citys. It also changes the viewheight of the page after a while. But i believe that it is a really small change that is needed to get that sorted. 

Overall I think you do a great job with keeping your code consistent and it shows when reading your code. 

The blue level is reached as it now and with a few changes you are at the red level!
Keep up the good work!

File: index.html [hunk 9]
```diff
@@ -0,0 +1,62 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link rel="stylesheet" href="style.css">
```
Comment (EmOlsz) [inline index.html:7]: Let's not forget about `./` in path to file

File: index.html [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+</head>
+
+<body>
+    <header class="location">
+        <input type="text" id="search" class="search-input" autocomplete="off" placeholder="Change location" />
+        <h1 id="name"></h1>
+        <img id="weatherPic" alt="current weather-icon" />
+
+        <h2 id="day" class="today">
+            <h2>
```
Comment (EmOlsz) [inline index.html:21]: Wrong indentation

File: script.js [hunk 11]
```diff
@@ -0,0 +1,122 @@
+import { API_KEY } from './api.js';
```
Comment (EmOlsz) [inline script.js:1]: Good job :)

File: script.js [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+let cityName = '';
+const city = document.getElementById('name');
+const displayDay = document.getElementById('days');
+const displayTemp = document.getElementById('temp');
+const displayType = document.getElementById('description');
+const displayIcon = document.getElementById('icon');
+
+//displays the geo location as default
+function geoLocate() {
+    if (navigator.geolocation) {
```
Comment (EmOlsz) [inline script.js:13]: Great achievement!

File: script.js [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+            getWeatherForecast(geoLocationForecastAPI)
+            getWeatherToday(geoLocationWeatherAPI)
+        });
+    } else {
+        alert("Geolocation is not supported by this browser.");
+    }
+};
+
+//Change city on input and keypress
+input.addEventListener('keypress', function(e) {
```
Comment (EmOlsz) [inline script.js:29]: It should rather be like this : `input.addEventListener('keypress', e => { ... }`

File: script.js [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+    fetch(url)
+        .then((response) => {
+            return response.json();
+        })
+        .then((currentWeather) => {
+            upDateWeather(currentWeather);
+        })
+        .catch((error) => {
+            city.innerHTML = ('Oops! Try again');
+
```
Comment (EmOlsz) [inline script.js:52]: We could also add some log to console about the error like that :

```
console.error(error);
```

As you can see, I used new `console.error()` type. It works the same as `console.log()`, but also adds red background to the message in Console tab indicating that something went wrong

File: script.js [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+            const filteredArray = forecastArray.list.filter(item => item.dt_txt.includes('12:00'));
+
+            const forecast = filteredArray.map((item) => {
+                const day = (new Date(item.dt * 1000)).toLocaleDateString("en-US", { weekday: "long" })
+                const temperature = (item.main.temp).toFixed(1);
+                const weatherType = item.weather[0].description;
+                const image = `<img src="https://openweathermap.org/img/wn/${item.weather[0].icon}@2x.png"/>`;
+                displayDay.innerHTML += `<p class="forecast">${day}&nbsp &nbsp${image}&nbsp &nbsp${temperature} °C</p>`;
+            });
+        });
```
Comment (EmOlsz) [inline script.js:119]: Missing `catch()`

File: script.js [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+};
+
+geoLocate()
```
Comment (EmOlsz) [inline script.js:122]: You seem to miss semicolons in some places

**Review by EmOlsz — CHANGES_REQUESTED** [review]
Hey Sara! Your code looks very nice, I'm impressed by amount of extra requirements :)
Some things can be improved, that's why I changed status to Changes requested. Please, provide changes and then I will accept it.
Moreover, I had problem with displaying your site - netlify link doesn't seem to work. Please, fix that as well :)

**Review by EmOlsz — APPROVED** [review]

**Pull request closed without merging**